Sources/NIOHTTP2/HTTP2Parser.swift

```diff
@@ -235,6 +237,17 @@ public final class HTTP2Parser: ChannelInboundHandler, ChannelOutboundHandler {
         self.reentrancyManager.markFlushPoint()
         self.reentrancyManager.process(ctx: ctx, self.process)
     }
+  
+    private func initializeSession(ctx: ChannelHandlerContext) {
+        self.session = NGHTTP2Session(mode: self.mode,
```

Comment:
This code should probably refuse to double-initialise, which means it should check whether `session` is `nil` before proceeding.

Reply:
done